src/components/TimeZonePicker.vue

```diff
@@ -0,0 +1,148 @@
+<!-- ADD_LICENSE_HEADER -->
+<template>
+	<div class="input-group">
+		<div class="input-group-prepend">
+			<b-button
+				@click="updateOffset"> 
+				UTC ±
+			</b-button>
+		</div>
+		<input
+			v-bind:value="timezoneString"
+			@input="validate"
+			@change="submitChange"
+			placeholder="[+-]hh:mm"
+			class="form-control" />
+	</div>
+</template>
+
+<script>
+export default {
+	name: 'timezonepicker',
+	model: {
+		prop: 'value',
+		event: 'input'
+	},
+	props: {
+		format: String,
+		value: String,
+		timezonedate: String
+	},
+	data() {
+		return {
+			internal_value: null,
+			initial_value: null,
+			isInitializing: true,
+			inputRegexp: /^[-+\d:]+$/,
+			timezoneRegexp: /^[+-]?(2[0-3]|[0-1]?\d)(:[0-5]?\d|:)?$/
+		}
+	},
+	computed: {
+		timezoneString: {
+			get() {
+				return this.internal_value
+			},
+			set(new_value) {
+				if (this.isInitializing) { return }
+				if (this.timezoneRegexp.test(new_value)) {
+					this.internal_value = new_value
+				}
+			}
+		},
+	},
+	methods: {
+		updateOffset() {
+			// this.date is in day.month.year format
+			// javascript Date expects it to month.day.year format
+			var [day, month, year] = this.timezonedate.split(".")
+			var monthIndex = month - 1
+			var local_offset = new Date(year, monthIndex, day).getTimezoneOffset()
+			if (isNaN(local_offset)) {
+				this.initial_value = null
+				this.internal_value = this.initial_value
+				this.$emit('input', this.initial_value)
+				return
+			}
+			// local_offset is -180 for Helsinki
+			// lets convert it to hh:mm format
+			var is_ahead_of_utc = local_offset < 0
+			var is_utc = local_offset === 0
+			var hours = String(Math.abs(local_offset/60)).padStart(2,"0")
+			var minutes = String(Math.abs(local_offset%60)).padStart(2,"0")
+			var symbol = ""
+			if (is_ahead_of_utc) {
+				symbol = "+"
+			} else if (is_utc) {
+				symbol = "+"
+			} else {
+				symbol = "-"
+			}
+			this.initial_value = symbol + hours + ":" + minutes
+			this.internal_value = this.initial_value
+			this.$emit('input', this.initial_value)
+		},
+		submitChange(event) {
```

Comment:
Is this the same logic as in the another component? Maybe share it with mixin?

Reply:
yes we should do that, but there is no time for a proper refactoring tour for the release